### PR TITLE
DOCS-2403, DOCS-2624, and DOCS-2622 Synthetics CI/CD Testing to CI/CD Integrations Navigation (Take 3)

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

We are changing "CI/CD Testing" to "CI/CD Integrations" and preparing the Synthetics CI/CD Integrations content update with a navigational update first. I replaced all English links for synthetics/cicd_testing with synthetics/cicd_integrations and added three Markdown files for new page content.

### Motivation

DOCS-2403, DOCS-2624, and DOCS-2622

### Preview

https://docs-staging.datadoghq.com/alai97/synthetics-cicd-testing-navigational-update/synthetics/cicd_integrations

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/synthetics/cicd_integrations/_index.md [hunk 1]
```diff
@@ -376,17 +376,17 @@ To learn how to get started using the testing tunnel, see the [Synthetics testin
...
 
-{{< img src="synthetics/ci/failed_test_result.png" alt="Failed Test Result" style="width:100%;">}}
+{{< img src="synthetics/cicd_integrations/failed_test_result.png" alt="Failed Test Result" style="width:100%;">}}
 
 #### In Datadog application
 
 You can also see your CI test results listed in the [CI Results Explorer][5] and on test details pages:
 
-{{< img src="synthetics/ci/ci_results_explorer/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
+{{< img src="synthetics/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
```
Comment (alai97) [inline content/en/synthetics/cicd_integrations/_index.md:389]:
```suggestion
{{< img src="synthetics/ci_results_explorer/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
```

File: content/en/synthetics/ci_results_explorer.md [hunk 2]
```diff
@@ -28,11 +28,11 @@ You can:
 * Keep track of the progress of your tests in CI pipelines.
 * Identify flaky tests to fix.
 
-{{< img src="synthetics/ci/ci_results_explorer/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
+{{< img src="synthetics/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
```
Comment (alai97) [inline content/en/synthetics/ci_results_explorer.md:31]:
```suggestion
{{< img src="synthetics/ci_results_explorer/ci_results_explorer.png" alt="CI Results Explorer" style="width:100%;">}}
```

**Review by ruthnaebeck — APPROVED** [review]
👍

File: config/_default/menus/menus.en.yaml [hunk 3]
```diff
@@ -2566,37 +2566,53 @@ main:
...
+  - name: Jenkins
+    url: /synthetics/cicd_integrations/jenkins
+    parent: synthetics_cicd_integrations
+    identifier: synthetics_cicd_integrations_jenkins
+    weight: 502
+  - name: GitHub Actions
+    url: /synthetics/cicd_integrations/github_actions
+    parent: synthetics_cicd_integrations
+    identifier: synthetics_cicd_integrations_github_actions
+    weight: 503
```
Comment (alai97) [inline config/_default/menus/menus.en.yaml:2588]:
```suggestion
  - name: GitHub Actions
    url: /synthetics/cicd_integrations/github_actions
    parent: synthetics_cicd_integrations
    identifier: synthetics_cicd_integrations_github_actions
    weight: 502
  - name: Jenkins
    url: /synthetics/cicd_integrations/jenkins
    parent: synthetics_cicd_integrations
    identifier: synthetics_cicd_integrations_jenkins
    weight: 503
```

**Review by alai97 — COMMENTED** [review]
The CI/CD Integrations PRs are ready for merge!

**Review by peculiarism — APPROVED** [review]

**Pull request merged**